### ISSUE TITLE
Question on tf.stop_gradient in dqn build_graph

### ISSUE THREAD

**Issue by Henry-bee** [openai/baselines#105]
Hey,

This question is regarding build_train function in DQN/build_graph.py. I notice that when you were computing td_error, you use tf.stop_gradient on target_Q values. Do you mind providing an explanation on this, What does stop_gradient do here, and how will it affect the overall loss function?

Regards
Henry

**Comment by Breakend** [comment]
The reason this is happening is so that gradients don't flow into the target Q function. Since 2 networks are being used here if you allow gradients to update the targetQ this becomes improper. This stop_gradient allows you to essentially treat targetQ as a constant that is the target update.

**Comment by olegklimov** [comment]
Correct!
